codejail_service/settings/base.py

```diff
@@ -26,218 +24,40 @@ def root(*path_fragments):
     root('conf', 'locale'),
 )
 
-
-# MEDIA CONFIGURATION
-# See: https://docs.djangoproject.com/en/dev/ref/settings/#media-root
-MEDIA_ROOT = root('media')
-
-# See: https://docs.djangoproject.com/en/dev/ref/settings/#media-url
-MEDIA_URL = '/media/'
-# END MEDIA CONFIGURATION
-
-
-# STATIC FILE CONFIGURATION
-# See: https://docs.djangoproject.com/en/dev/ref/settings/#static-root
-STATIC_ROOT = root('assets')
-
-# See: https://docs.djangoproject.com/en/dev/ref/settings/#static-url
-STATIC_URL = '/static/'
-
-# See: https://docs.djangoproject.com/en/dev/ref/contrib/staticfiles/#std:setting-STATICFILES_DIRS
-STATICFILES_DIRS = (
-    root('static'),
-)
-
-# TEMPLATE CONFIGURATION
-# See: https://docs.djangoproject.com/en/3.2/ref/settings/#templates
-TEMPLATES = [
-    {
-        'BACKEND': 'django.template.backends.django.DjangoTemplates',
-        'APP_DIRS': True,
-        'DIRS': (
-            root('templates'),
-        ),
-        'OPTIONS': {
-            'context_processors': (
-                'django.contrib.auth.context_processors.auth',
-                'django.template.context_processors.debug',
-                'django.template.context_processors.i18n',
-                'django.template.context_processors.media',
-                'django.template.context_processors.request',
-                'django.template.context_processors.static',
-                'django.template.context_processors.tz',
-                'django.contrib.messages.context_processors.messages',
-                'codejail_service.apps.core.context_processors.core',
-            ),
-            'debug': True,  # Django will only display debug pages if the global DEBUG setting is set to True.
-        }
-    },
-]
-# END TEMPLATE CONFIGURATION
-
-
-# COOKIE CONFIGURATION
-# The purpose of customizing the cookie names is to avoid conflicts when
-# multiple Django services are running behind the same hostname.
-# Detailed information at: https://docs.djangoproject.com/en/dev/ref/settings/
-SESSION_COOKIE_NAME = 'codejail_service_sessionid'
-CSRF_COOKIE_NAME = 'codejail_service_csrftoken'
-LANGUAGE_COOKIE_NAME = 'codejail_service_language'
-# END COOKIE CONFIGURATION
-
-CSRF_COOKIE_SECURE = False
-CSRF_TRUSTED_ORIGINS = []
-
-# AUTHENTICATION CONFIGURATION
-LOGIN_URL = '/login/'
-LOGOUT_URL = '/logout/'
-
-AUTH_USER_MODEL = 'core.User'
-
-AUTHENTICATION_BACKENDS = (
-    'auth_backends.backends.EdXOAuth2',
-    'django.contrib.auth.backends.ModelBackend',
-)
-
-ENABLE_AUTO_AUTH = False
-AUTO_AUTH_USERNAME_PREFIX = 'auto_auth_'
-
-SOCIAL_AUTH_STRATEGY = 'auth_backends.strategies.EdxDjangoStrategy'
-
-# Set these to the correct values for your OAuth2 provider (e.g., LMS)
-SOCIAL_AUTH_EDX_OAUTH2_KEY = 'replace-me'
-SOCIAL_AUTH_EDX_OAUTH2_SECRET = 'replace-me'
-SOCIAL_AUTH_EDX_OAUTH2_URL_ROOT = 'replace-me'
-SOCIAL_AUTH_EDX_OAUTH2_LOGOUT_URL = 'replace-me'
-BACKEND_SERVICE_EDX_OAUTH2_KEY = 'replace-me'
-BACKEND_SERVICE_EDX_OAUTH2_SECRET = 'replace-me'
-
-JWT_AUTH = {
```

Comment:
Is the eduNEXT version entirely without auth? Is it simply trusting the behavior of its caller?

Reply:
Correct -- theirs is an unauthenticated service as well. I originally wanted to protect ours behind JWT auth but decided it wasn't worth it; anyone can already make calls to the service as they wish by submitting answers to certain course problems. And we can just decline to expose a direct API to the internet anyhow.